src/providers/core/mod.rs

```diff
@@ -263,13 +262,29 @@ impl ProviderBuilder {
             provider_info_vec.push(provider_info);
         }
 
-        let crate_version: Version = Version::from_str(version!()).map_err(|e| {
-            format_error!("Error parsing the crate version", e);
+        let crate_version: std::result::Result<Vec<u32>, ParseIntError> = env!("CARGO_PKG_VERSION")
+            .split('.')
+            .map(|v| v.parse())
+            .collect();
```

Comment:
This is mostly fine, but in the case of release candidates (for example [this](https://github.com/parallaxsecond/parsec/releases/tag/1.1.0-rc2)) it (probably?) won't work, IIUC. I'd prefer avoiding regular expressions, maybe discard everything after the first `-` if it exists? That would also take care of any alpha/beta versioning as well.

Reply:
Ah, nevermind, it's parsing the version in `Cargo.toml`, and that one is always in the correct format (I'd hope), disregard this.